airbyte/sources/util.py

```diff
@@ -115,6 +117,28 @@ def get_source(
             ),
         )
 
+    if docker_executable:
+        version = version or "latest"
+
+        temp_dir = tempfile.gettempdir()
+        return Source(
+            name=name,
+            config=config,
+            streams=streams,
+            executor=DockerExecutor(
+                name=name,
+                executable=[
+                    "docker",
+                    "run",
+                    "--rm",
+                    "-i",
+                    "--volume",
+                    f"{temp_dir}:{temp_dir}",
+                    f"airbyte/{name}:{version}",
+                ],
+            ),
```

Comment:
Following from the above, I think we can take the image name (and optional version) as input, in place of `f"airbyte/{name}:{version}"`.

And then from there, I wonder if we can reduce the constructor to only require the image name? Or maybe also the local temp_dir to use for storing docker volume files?

Reply:
Yes that makes sense to me! No need to take in the entire docker run command and it's arguments as those are constants. For temp_dir I think we should keep that fixed as it needs to match the PyAirbyte temp_dir [here](https://github.com/airbytehq/PyAirbyte/blob/main/airbyte/_util/temp_files.py#L23) unless there is a way for overriding that which I am not seeing?